src/main/java/com/strandls/dataTable/util/LogActivities.java

```diff
@@ -0,0 +1,49 @@
+package com.strandls.dataTable.util;
+
+
+import javax.inject.Inject;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.strandls.activity.controller.ActivitySerivceApi;
+import com.strandls.activity.pojo.ActivityLoggingData;
+import com.strandls.activity.pojo.MailData;
+import com.strandls.dataTable.Headers;
+
+/**
+ * @author Abhishek Rudra
+ *
+ */
+public class LogActivities {
+
+	private final Logger logger = LoggerFactory.getLogger(LogActivities.class);
+
+	@Inject
+	private ActivitySerivceApi activityService;
+
+	@Inject
+	private Headers headers;
+
+	public void LogActivity(String authHeader, String activityDescription, Long rootObjectId,
+			Long subRootObjectId, String rootObjectType, Long activityId, String activityType, MailData mailData) {
+
+		try {
+			ActivityLoggingData activityLogging = new ActivityLoggingData();
+			activityLogging.setActivityDescription(activityDescription);
+			activityLogging.setActivityId(activityId);
+			activityLogging.setActivityType(activityType);
+			activityLogging.setRootObjectId(rootObjectId);
+			activityLogging.setRootObjectType(rootObjectType);
+			activityLogging.setSubRootObjectId(subRootObjectId);
+			activityLogging.setMailData(mailData);
+			activityService = headers.addActivityHeaders(activityService, authHeader);
+			activityService.logActivity(activityLogging);
```

Comment:
Seems this is observation activity, shouldnt it be data table activity

Reply:
changed with datatable activity